src/item.cpp

```diff
@@ -2371,6 +2376,35 @@ bool item::has_flag( const std::string &f ) const
     return ret;
 }
 
+bool item::has_property (const std::string& prop) const {
+   return type->properties.find(prop) != type->properties.end();
+}
+
+std::string item::get_property (const std::string &prop, const std::string& def) const
```

Comment:
Looks like only get_property( std::string&, std::string) and get_property(std::string, int) are used, please remove the others.
Also the API where you determine the return value based on the default value is very odd, please just distinguish between them with names, e.g. get_property_int()